binding.cc

```diff
@@ -1036,22 +1039,77 @@ NAPI_METHOD(db_get_many) {
         return rocksdb::Status::OK();
       },
       [=](auto& state, auto env, auto& argv) {
-        argv.resize(3);
+        argv.resize(2);
 
-        if (state.sizes.size() > 0) {
-          auto sizes = std::make_unique<std::vector<int32_t>>(std::move(state.sizes));
-          NAPI_STATUS_RETURN(napi_create_external_buffer(env, sizes->size() * 4, sizes->data(), Finalize<std::vector<int32_t>>, sizes.get(), &argv[1]));
-          sizes.release();
-        } else {
-          NAPI_STATUS_RETURN(napi_get_undefined(env, &argv[1]));
-        }
+        NAPI_STATUS_RETURN(napi_create_array_with_length(env, state.sizes.size(), &argv[1]));
+
+        if (valueEncoding == Encoding::Buffer) {
+          napi_value arraybuffer;
+          {
+            // We create one ArrayBuffer and then slice it into Buffer(s) to avoid the
+            // overhead of registering lots of finalizers which is super slow.
+            // https://github.com/nodejs/node/issues/53804
+            auto data = std::make_unique<std::vector<uint8_t>>(std::move(state.data));
+            NAPI_STATUS_RETURN(napi_create_external_arraybuffer(
```

Comment:
Just a note that in Node.js implementation, `napi_create_external_arraybuffer` is an alias for `napi_create_external_buffer`: https://github.com/nodejs/node/blob/main/src/js_native_api_v8.cc#L3027-L3041